grype/presenter/cyclonedx/document.go

```diff
@@ -0,0 +1,149 @@
+package cyclonedx
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+
+	"github.com/anchore/grype/grype/match"
+	"github.com/anchore/grype/grype/result"
+	"github.com/anchore/grype/grype/vulnerability"
+	"github.com/anchore/syft/syft/pkg"
+	syftCDX "github.com/anchore/syft/syft/presenter/cyclonedx"
+	"github.com/google/uuid"
+)
+
+// Source: https://github.com/CycloneDX/specification
+
+// Document represents a CycloneDX Vulnerability Document.
+type Document struct {
+	XMLName       xml.Name               `xml:"bom"`
+	XMLNs         string                 `xml:"xmlns,attr"`
+	Version       int                    `xml:"version,attr"`
+	SerialNumber  string                 `xml:"serialNumber,attr"`
+	Components    []Component            `xml:"components"`
```

Comment:
I lied to you the last we spoke, I learned that `components>component` is necessary for marshaling as well. This hints to the XML lib to behave like so:
```xml
<components>
    <component>thing1</component>
    <component>thing2</component>
</components>
```
instead of like so:
```xml
<component>thing1</component>
<component>thing2</component>
```